src/framework/entity.js

```diff
@@ -379,6 +379,36 @@ class Entity extends GraphNode {
         });
     }
 
+    /**
+     * Search the entity and all of its descendants for the first script instance of specified type.
+     *
+     * @param {string|Class<import('./script/script-type.js').ScriptType>} nameOrType - The name or type of {@link ScriptType}.
+     * @returns {import('./script/script-type.js').ScriptType} A script instance of specified type, if the entity or any of its descendants
+     * has one. Returns undefined otherwise.
+     * @example
+     * // Get the first found "playerController" instance in the hierarchy tree that starts with this entity
+     * var controller = entity.findScript("playerController");
+     */
+    findScript(nameOrType) {
+        const entity = this.findOne(node => node.c && node.c.script && node.c.script.has(nameOrType));
+        return entity && entity.c.script.get(nameOrType);
```

Comment:
could probably be shortened to
```suggestion
        const entity = this.findOne(node => node.c?.script?.has(nameOrType));
        return entity?.c.script.get(nameOrType);
```

Reply:
entity can be undefined, so `entity?.` should be used